In [1]:
import mysql.connector as mydbconnection
from mysql.connector import Error

def connect():
    """ Connect to MySQL database """
    conn = None
    try:
        conn = mydbconnection.connect(host='localhost',
                                      database='classicmodels',
                                             user='root',
                                             password='password')
        if conn.is_connected():
            print('Connected to MySQL database')

    except Error as e:
        print(e) # print error

    finally:
        if conn is not None and conn.is_connected():
            conn.close() # close connection
            print('Connection closed')


if __name__ == '__main__':
    connect()


2003 (HY000): Can't connect to MySQL server on 'localhost:3306' (111)


In [4]:
conn = mydbconnection.connect(database='classicmodels',
                                             user='root',
                                             password='password')
if conn.is_connected():
    print('Connected to MySQL database')
else:
    print('No connection')

DatabaseError: 2003 (HY000): Can't connect to MySQL server on '127.0.0.1:3306' (111)

#### Note:
- If the username or password is not valid, you will get the following error:
    - `1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)`

- In case the MySQL is not available, you will get the following error:
    - `2003: Can't connect to MySQL server on 'localhost:3306' (10061 No connection could be made because the target machine actively refused it)`